### ISSUE TITLE
strip_typescript returns raw TypeScript for a file with an istanbul ignore file pragma

### ISSUE THREAD

**Issue by BartWaardenburg** [fallow-rs/oxc-coverage-instrument#172]
With `strip_typescript` enabled, a TypeScript file carrying `/* istanbul ignore file */` is returned verbatim, so the output still contains type annotations and cannot be executed.

### Reproduction

`ok.ts`:

```ts
export const add = (a: number, b: number): number => a + b;
```

```
oxc-coverage-instrument ok.ts
```

Output is JavaScript, as documented:

```js
export const add = (a, b) => {
	return a + b;
};
```

Now `bug.ts`, the same file with the pragma:

```ts
/* istanbul ignore file */
export const add = (a: number, b: number): number => a + b;
```

```
oxc-coverage-instrument bug.ts
```

Output:

```ts
/* istanbul ignore file */
export const add = (a: number, b: number): number => a + b;
```

### Expected

The ignore-file pragma should suppress instrumentation, not the TypeScript strip. The output should still be executable JavaScript, just without counters.

A build pipeline that relies on `strip_typescript` receives un-executable output for every ignored `.ts` file, and no error is raised.

### Cause

In `crates/oxc_coverage_instrument/src/instrument.rs`, `instrument()` returns `empty_coverage_result(filename, source, ..)` for an ignore-file source before `prepare_scoping` and the TypeScript strip pass run, so `code` is the untouched input.

### Version

0.10.3
